packages/app/src/app/pages/NewDashboard/Components/Selection/ContextMenu.tsx

```diff
@@ -41,16 +44,38 @@ export const ContextMenu: React.FC<IContextMenuProps> = ({
   selectedIds,
   sandboxes,
   folders,
+  repos,
   setRenaming,
   createNewFolder,
   createNewSandbox,
 }) => {
   if (!visible) return null;
 
   const selectedItems: Array<
-    DashboardFolder | DashboardSandbox | DashboardTemplate
+    | DashboardFolder
+    | DashboardSandbox
+    | DashboardTemplate
+    | DashboardRepo
+    | DashboardNewMasterBranch
   > = selectedIds.map(id => {
     if (id.startsWith('/')) {
+      if (repos.length) {
+        const repo = repos.find(f => '/' + f.name === id);
+        return { type: 'repo', ...repo };
+      }
+
+      if (id.includes('/github')) {
```

Comment:
Is there a more explicit way we can check here? To make sure nothing goes wrong? Like checking for `id.startsWith(` instead maybe?

Reply:
changed